File: 70. 爬楼梯/swift/main/main/main.swift
//
//  main.swift
//  main
//
//  Created by ByteDance on 4/28/25.
//

import Foundation

class Solution {
    var dp = [Int: Int]()
    func climbStairs(_ n: Int) -> Int {
        if n <= 3 {
            return n
        }
        dp[1] = 1
        dp[2] = 2
        dp[3] = 3
        for i in 4 ... n {
            dp[i] = dp[i - 1]! + dp[i - 2]!
        }
        return dp[n]!
    }
}

File: 70. 爬楼梯/rust/main/src/main.rs
struct Solution {}

impl Solution {
    pub fn climb_stairs(n: i32) -> i32 {
        let mut list: Vec<i32> = vec![0; (n+1) as usize];
        if n == 0 {
            return 1
        }
        if n == 1 {
            return 1
        }
        list[0] = 1;
        list[1] = 1;
        for i in 2 ..= n {
            list[i as usize] = list[(i-1) as usize] + list[(i-2) as usize]
        }
        return list[n as usize]
    }
}

fn main() {
    println!("Hello, world!");
    println!("{}", Solution::climb_stairs(2));
}
